File: src/main/java/com/anod/appwatcher/market/SearchEndpoint.java
package com.anod.appwatcher.market;

import android.content.Context;

import com.google.android.finsky.api.DfeUtils;
import com.google.android.finsky.api.model.DfeModel;
import com.google.android.finsky.api.model.DfeSearch;

/**
 * @author alex
 * @date 2015-02-21
 */
public class SearchEndpoint extends PlayStoreEndpointBase {

    private static final int BACKEND_ID = 3;
    private String mQuery;
    private boolean mAutoLoadNextPage;

    public SearchEndpoint(Context context, boolean autoLoadNextPage) {
        super(context);
        mAutoLoadNextPage = autoLoadNextPage;
    }

    public String getQuery() {
        return mQuery;
    }

    public SearchEndpoint setQuery(String query) {
        mQuery = query;
        return this;
    }

    public DfeSearch getData() {
        return (DfeSearch)mDfeModel;
    }

    @Override
    public void reset() {
        if (getData() != null) {
            getData().resetItems();
            super.reset();
        }
    }

    public int getCount() {
        if (mDfeModel == null) {
            return 0;
        }
        return getData().getCount();
    }


    @Override
    protected void executeAsync() {
        getData().startLoadItems();
    }

    @Override
    protected void executeSync() {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    protected DfeModel createDfeModel() {
        String searchUrl = DfeUtils.formSearchUrl(mQuery, BACKEND_ID);
        return new DfeSearch(mDfeApi, mQuery, searchUrl, mAutoLoadNextPage, AppDetailsFilter.createPredicate());
    }
}

File: src/main/java/com/anod/appwatcher/market/SearchEndpoint.kt
package com.anod.appwatcher.market

import android.content.Context

import com.google.android.finsky.api.DfeUtils
import com.google.android.finsky.api.model.DfeModel
import com.google.android.finsky.api.model.DfeSearch

/**
 * @author alex
 * *
 * @date 2015-02-21
 */
class SearchEndpoint(context: Context, private val mAutoLoadNextPage: Boolean) : PlayStoreEndpointBase(context) {
    var query: String = ""

    var searchData: DfeSearch?
        get() = data as? DfeSearch
        set(value) {
            this.data = value
        }

    override fun reset() {
        searchData?.resetItems()
        super.reset()
    }

    val count: Int
        get() = searchData?.count ?: 0

    override fun executeAsync() {
        searchData?.startLoadItems()
    }

    override fun executeSync() {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun createDfeModel(): DfeModel {
        val searchUrl = DfeUtils.formSearchUrl(query, BACKEND_ID)
        return DfeSearch(dfeApi, query, searchUrl, mAutoLoadNextPage, AppDetailsFilter.predicate)
    }

    companion object {
        private val BACKEND_ID = 3
    }
}
